test/robot/scheduler/replay.go

```diff
@@ -24,12 +24,23 @@ import (
 	"github.com/google/gapid/test/robot/replay"
 )
 
-func (s schedule) doReplay(ctx context.Context, t *monitor.Trace, tools *build.ToolSet) error {
+func (s schedule) doReplay(ctx context.Context, t *monitor.Trace,
+	tools *build.ToolSet, androidTools *build.AndroidToolSet) error {
+	gapirdev := "host"
 	if !s.worker.Supports(job.Replay) {
 		return nil
 	}
+	if t.Input.Hints.API == "vulkan" {
```

Comment:
I'd prefer it if this was explicit (for example a `ReportDevice` field) instead of using magic string comparisons. GLES will want to do this eventually.

Reply:
Do you mean to add a `ReportDevice` field to Trace Action (robot/trace/trace.proto), as the report device can only be determined once the trace is done?

I'm thinking to make an enum type for "vulkan" and "gles", I agree this magic string comparison is ugly and not reliable.

Reply:
Well, I was hoping that input provider could know whether it should be doing the report on the host or target device. I don't like scattering API specific logic around the codebase.
GLES will need to support on-device replay too. I don't want to search through random places to fix up the rules on what API can do what.

Reply:
So we want, at the time of uploading a subject, specify 'host', 'android' or 'serial number' as the device for trace/replay/report? Like what we have for gapit trace/report/replay?
And we can change default selection based on the given api only in the cmd/robot/subject.

Is this what you expect to have?

Reply:
We could do that. I don't mind it being slightly more automatic, so long as there is a centralized place for the rules.

Reply:
So in 6807245, I put all those rules to scheduler.go. Didn't change the uploading command flags and anything else.